sonic_platform_base/sonic_xcvr/api/public/cmis.py

```diff
@@ -16,5 +16,270 @@ def __init__(self, xcvr_eeprom):
+                               'txbiashighalarm',  'txbiashighwarning',
+                               'txbiaslowalarm',   'txbiaslowwarning'
+                              ]
+        threshold_info_dict = dict.fromkeys(threshold_info_keys, 'N/A')
+
+        thresh_support = self.get_transceiver_thresholds_support()
+        if thresh_support is None:
+            return None
+        if not thresh_support:
+            return threshold_info_dict
+        thresh = self.xcvr_eeprom.read(consts.THRESHOLDS_FIELD)
+        if thresh is None:
+            return None
+
+        return {
+            "temphighalarm": float("{:.3f}".format(thresh[consts.TEMP_HIGH_ALARM_FIELD])),
+            "templowalarm": float("{:.3f}".format(thresh[consts.TEMP_LOW_ALARM_FIELD])),
+            "temphighwarning": float("{:.3f}".format(thresh[consts.TEMP_HIGH_WARNING_FIELD])),
+            "templowwarning": float("{:.3f}".format(thresh[consts.TEMP_LOW_WARNING_FIELD])),
+            "vcchighalarm": float("{:.3f}".format(thresh[consts.VOLTAGE_HIGH_ALARM_FIELD])),
+            "vcclowalarm": float("{:.3f}".format(thresh[consts.VOLTAGE_LOW_ALARM_FIELD])),
+            "vcchighwarning": float("{:.3f}".format(thresh[consts.VOLTAGE_HIGH_WARNING_FIELD])),
+            "vcclowwarning": float("{:.3f}".format(thresh[consts.VOLTAGE_LOW_WARNING_FIELD])),
+            "rxpowerhighalarm": float("{:.3f}".format(thresh[consts.RX_POWER_HIGH_ALARM_FIELD])),
+            "rxpowerlowalarm": float("{:.3f}".format(self.mw_to_dbm(thresh[consts.RX_POWER_LOW_ALARM_FIELD]))),
+            "rxpowerhighwarning": float("{:.3f}".format(self.mw_to_dbm(thresh[consts.RX_POWER_HIGH_WARNING_FIELD]))),
+            "rxpowerlowwarning": float("{:.3f}".format(self.mw_to_dbm(thresh[consts.RX_POWER_LOW_WARNING_FIELD]))),
+            "txpowerhighalarm": float("{:.3f}".format(self.mw_to_dbm(thresh[consts.TX_POWER_HIGH_ALARM_FIELD]))),
+            "txpowerlowalarm": float("{:.3f}".format(self.mw_to_dbm(thresh[consts.TX_POWER_LOW_ALARM_FIELD]))),
+            "txpowerhighwarning": float("{:.3f}".format(self.mw_to_dbm(thresh[consts.TX_POWER_HIGH_WARNING_FIELD]))),
+            "txpowerlowwarning": float("{:.3f}".format(self.mw_to_dbm(thresh[consts.TX_POWER_LOW_WARNING_FIELD]))),
+            "txbiashighalarm": float("{:.3f}".format(thresh[consts.TX_BIAS_HIGH_ALARM_FIELD])),
+            "txbiaslowalarm": float("{:.3f}".format(thresh[consts.TX_BIAS_LOW_ALARM_FIELD])),
+            "txbiashighwarning": float("{:.3f}".format(thresh[consts.TX_BIAS_HIGH_WARNING_FIELD])),
+            "txbiaslowwarning": float("{:.3f}".format(thresh[consts.TX_BIAS_LOW_WARNING_FIELD]))
+        }
+
+    def get_module_temperature(self):
+        if not self.get_temperature_support():
+            return 'N/A'
+        temp = self.xcvr_eeprom.read(consts.TEMPERATURE_FIELD)
+        if temp is None:
+            return None
+        return float("{:.3f}".format(temp))
+ 
+    def get_voltage(self):
```

Comment:
Voltage seems to be returning with incorrect floating point value in new sff8436. Please check the same in CMIS too.
**With earlier method:**
>>> c.get_sfp(2).get_voltage()
3.3804
>>> c.get_sfp(2).get_transceiver_threshold_info()
{'temphighalarm': '75.0000C', 'temphighwarning': '70.0000C', 'templowalarm': '-5.0000C', 'templowwarning': '0.0000C', 'vcchighalarm': '3.6300Volts', 'vcchighwarning': '3.4650Volts', 'vcclowalarm': '2.9700Volts', 'vcclowwarning': '3.1350Volts', 'rxpowerhighalarm': '3.3999dBm', 'rxpowerhighwarning': '2.4000dBm', 'rxpowerlowalarm': '-13.5067dBm', 'rxpowerlowwarning': '-9.5001dBm', 'txpowerhighalarm': 'N/A', 'txpowerhighwarning': 'N/A', 'txpowerlowalarm': 'N/A', 'txpowerlowwarning': 'N/A', 'txbiashighalarm': '15.0000mA', 'txbiashighwarning': '14.0000mA', 'txbiaslowalarm': '2.0000mA', 'txbiaslowwarning': '3.0000mA'}
>>>

**With refactored method:**
>>> c.get_sfp(2).get_voltage()
**33.816**
>>> c.get_sfp(2).get_transceiver_threshold_info()
{'temphighalarm': 75.0, 'templowalarm': -5.0, 'temphighwarning': 70.0, 'templowwarning': 0.0, **'vcchighalarm': 36.3, 'vcclowalarm': 29.7, 'vcchighwarning': 34.65, 'vcclowwarning': 31.35,** 'rxpowerhighalarm': 13.4, 'rxpowerlowalarm': -3.507, 'rxpowerhighwarning': 12.4, 'rxpowerlowwarning': 0.5, 'txpowerhighalarm': 'N/A', 'txpowerlowalarm': 'N/A', 'txpowerhighwarning': 'N/A', 'txpowerlowwarning': 'N/A', 'txbiashighalarm': 15.0, 'txbiaslowalarm': 2.0, 'txbiashighwarning': 14.0, 'txbiaslowwarning': 3.0}

Reply:
Requesting you to compare the output retrieved with OOM diag tool too.

Reply:
Fixed, thanks for catching this.